stdlib/LinearAlgebra/test/tridiag.jl

```diff
@@ -82,7 +82,7 @@ end
             @test TT == Matrix(TT)
             @test TT.dl === y
             @test TT.d  === x
-            @test TT.du === y
+            @test TT.du == y
```

Comment:
Should we document which off-diagonal is aliased to the argument of the constructor and which one may be just a copy?

Reply:
I feel we should document that the off diagonals shouldn't be aliased, and leave the de-aliasing as an implementation detail.

Reply:
Hello, as a Julia beginner and new user, I feel I might prefer to have a warning here (and not an error as earlier) if I alias `dl` and `du`. The reason is that, in Julia, I always expect matrix assignments and calls to reuse the same memory. Silent unaliasing of `dl` and `du` would be (from my point of view) unexpected behaviour.

Reply:
I've added a comment to the docstring explaining that `du` is copied if aliasing is detected.

Reply:
Thank you very much, @jishnub. I'm not sure if it makes sense, in my comment above, I was actually referring to a runtime warning that would seem most correct to me as a beginner.

Reply:
I think runtime warnings may ruin performance. Moreover, we have the hint in the documentation, so it's not exactly "silent".